alvr/client_openxr/src/stream.rs

```diff
@@ -144,27 +147,33 @@ impl StreamContext {
             None
         };
 
+        let target_resolution = if let Some(upscaling) = config.upscaling.clone() {
+            scale_resolution(config.view_resolution, upscaling)
+        } else {
+            config.view_resolution
+        };
         let format = graphics::swapchain_format(&gfx_ctx, &xr_session, config.enable_hdr);
 
         let swapchains = [
             graphics::create_swapchain(
                 &xr_session,
                 &gfx_ctx,
-                config.view_resolution,
+                target_resolution,
                 format,
                 foveation_profile.as_ref(),
             ),
             graphics::create_swapchain(
                 &xr_session,
                 &gfx_ctx,
-                config.view_resolution,
+                target_resolution,
                 format,
                 foveation_profile.as_ref(),
             ),
         ];
 
         let renderer = StreamRenderer::new(
             gfx_ctx,
+            // pass config.view_resolution as streamrenderer needs to perform own operations with upscaler
```

Comment:
As i suggested, you should pass also the precomputed target_view_resolution here